src/main/java/mods/railcraft/common/blocks/multi/TileMultiBlock.java

```diff
@@ -454,6 +454,7 @@ public final boolean isStructureValid() {
     @Override
     public final @Nullable TileMultiBlock getMasterBlock() {
         if (masterBlock != null && !isStructureValid()) {
+            masterBlock.isMaster = false;
```

Comment:
Don't like this. Its not the slaves job to determine whether a master is a master.

Reply:
So should I call `isStructureValid()` in `isMaster()` check?

Reply:
Updated. Added a contract to call both methods if work should be done.